rfcs/text/pe/pe.yml

```diff
@@ -0,0 +1,193 @@
+---
+- name: pe
+  type: group
+  fields:
+    - name: main_icon
+      level: extended
+      type: object
+      description: >
+        Hashes of embedded program icon.
+      fields:
+        - name: dhash
+          level: extended
+          type: keyword
+          description: >
+            Difference Hash (dhash) for a given PE file.
+          example: b806e17c8e330d82
+
+        - name: md5
+          level: extended
+          type: keyword
+          description: >
+            MD5 hash of raw icon data.
+          example: 6d1cae6272afbb88876ed6476b990d8c
+
+    - name: debug
```

Comment:
this should be a structured set of fields rather than a keyword, no?

Reply:
yeah, and I think the same could be said for
* import_list
* sections
* resource_details
* resource_languages
* resource_types
* packers

How do I represent that properly?

Reply:
or do you mean?
```
    - name: debug
      level: extended
      type: object
      description: Debug information, if present
      fields:
        - name: offset
           level: extended
           type: keyword
           description: description
           example: 1296336
...
```

Reply:
or do you mean?
```
    - name: debug
      level: extended
      type: object
      description: Debug information, if present
      fields:
        - name: offset
           level: extended
           type: keyword
           description: description
           example: 1296336
...
```

Reply:
so, there are basically two mechanisms AFAIK for making lists of structured sets of fields -- you'd declare the fields as either `nested` or `flattened`. [`nested`](https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html) causes some performance penalties due to the way it indexes additional documents and has some limitations on the number of nested objects that a single document can contain (1000 by default, though I doubt we'd hit that). Alternatively you can use [`flattened`](https://www.elastic.co/guide/en/elasticsearch/reference/current/flattened.html) which is an Elastic-licensed only thiing. It keeps the mapping sizes under control, but has the limitation that it essentially makes each subfield a `keyword` field.

You'll likely want to go through each of the fields that can be a "list of objects" and assess whether it makes sense to be `nested` or `flattened` or just a list of keywords (maybe `packers` if you're only putting the name of the packer there?).

Reply:
Opted for `nested`; LMK if you think that `flattened` makes more sense.